src/shared/upload-file.js

```diff
@@ -62,3 +64,47 @@ export function selectAndUploadFile(notebookID, successCallback = () => {}) {
       .then(output => successCallback(output));
   });
 }
+
+export function valueToFile(data, fileName) {
+  return new File([data], fileName);
+}
+
+export function makeFormData(notebookID, data, fileName) {
+  const file = valueToFile(data, fileName);
+  const formData = new FormData();
+  formData.append(
+    "metadata",
+    JSON.stringify({
+      filename: file.name,
+      notebook_id: notebookID
+    })
+  );
+  formData.append("file", file);
+  return formData;
+}
+
+export function saveFileToServer(
```

Comment:
This file needs to be refactored and generalized to just be something like `file-operations.js`.